Multisig/Logic/Ethereum/BlockchainDomainManager.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  BlockchainDomainManager.swift
+//  Multisig
+//
+//  Created by Johnny Good on 1/20/21.
+//  Copyright © 2021 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+import UnstoppableDomainsResolution
+
+class BlockchainDomainManager {
+    
+    let ens: ENS;
+    let resolution: Resolution;
+    
+    init() {
+        ens = ENS(registryAddress: "0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e");
+        self.resolution = try! Resolution();
+    }
+    
+    func resolveUD(_ domain: String) throws -> Address {
+        var address: String = "";
+        var err: Error? = nil;
+        
+        let dispatchGroup = DispatchGroup();
+        dispatchGroup.enter();
+        resolution.addr(domain: domain, ticker: "eth") { result in
+            switch result {
+                case .success(let returnValue):
+                    address = returnValue;
+                case .failure(let error):
+                  err = error;
+          }
+            dispatchGroup.leave();
+        };
+        dispatchGroup.wait();
+        
+        guard err == nil else {
+            if (err is ResolutionError) {
+                throw self.throwCorrectUdError(err as! ResolutionError, domain);
+            }
+            throw err!;
+        }
+        return Address(address)!;
```

Comment:
we could use `Address(exactly: address)` here to remove force unwrap

Reply:
however, if the `address` is not guaranteed to be an address String, there's an option to use `try Address(from: address)`

Reply:
Alternative is to use `try Address(from: address)` if there's a possibility that the address string is not an Ethereum address